packages/svelte/src/compiler/phases/3-transform/client/utils.js

```diff
@@ -311,3 +311,44 @@ export function create_derived_block_argument(node, context) {
 export function create_derived(state, arg) {
 	return b.call(state.analysis.runes ? '$.derived' : '$.derived_safe_equal', arg);
 }
+
+/**
+ * @param {import('./types.js').ComponentClientTransformState} state
+ * @param {string} quasi_to_add
+ */
+export function push_template_quasi(state, quasi_to_add) {
+	const { quasi } = state.template;
+	if (quasi.length === 0) {
+		quasi.push(quasi_to_add);
+		return;
+	}
+	quasi[quasi.length - 1] = quasi[quasi.length - 1].concat(quasi_to_add);
```

Comment:
how does this work exactly, it looks like quasi.length can never exceed 1 at index 0 there is another array that gets concatenated instead of pushed to?

Reply:
He's pushing a new quasi after each expression. Basically as long as you only add quasis you are just concatenating the new quasi to the old string but when you add a new expression there will be a new empty string in `quasi` and every new quasi will be concat to that.